### PR TITLE
storage/imx6ull-flashsrv: a new version uses libstorage library and flashdev module

### PR DESCRIPTION
**Pull request by HBuczynski**
## Description

The majority of the code has been rewritten. The user interface to the server is the same.

**NOTE for reviewers:**
- in the previous implementation the failure of a `create_dev` function has been treated as a non-critical error. Should be the same here? In the code, there is a note in the `TODO` comment.
- `flashsrv_write` had a special feature - erasing memory if `data` is passed as a `NULL`. Is it a feature or a bug? In my opinion it is a bug.

## Motivation and Context

DTR-102

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imx6ull, DTR project).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: storage/imx6ull-flash/flashsrv.c [hunk 1]
```diff
@@ -1,637 +1,372 @@
...
+	switch (type) {
+		case atSize:
+			*attr = (off_t)strg->size;
+			break;
 
-		if (err < 0) {
-			LOG_ERROR("error in readraw(): %d", err);
-			err = -EIO;
+		case atDev:
+			*attr = (off_t)strg->start;
```
Comment (agkaminski) [inline storage/imx6ull-flash/flashsrv.c:344]: I am pretty sure that's not what atDev is for

File: storage/imx6ull-flash/flashsrv.c [hunk 2]
```diff
@@ -665,166 +400,250 @@ static void flashsrv_devCtrl(msg_t *msg)
...
 
-		msgRespond(port, &msg, rid);
+	/* mount / symlink rootfs partition */
+	err = storage_mountfs(storage_get(rootfsID), fs, NULL, 0, &oid);
+	if (err < 0) {
+		LOG_ERROR("failed to mount a filesystem - %s: %d\n", fs, err);
+		return err;
 	}
+
+	portRegister(oid.port, "/", &oid);
```
Comment (agkaminski) [inline storage/imx6ull-flash/flashsrv.c:489]: Nitpick - no retval check

File: storage/imx6ull-flash/flashsrv.c (hunk 1 above)
Comment (Maxez) [reply storage/imx6ull-flash/flashsrv.c:344]: It's used in `psd` - not sure why: [link](https://github.com/phoenix-rtos/phoenix-rtos-utils/blob/a46e92d6e5002941b541d14b019d6139f2c18cc5/psd/imx6ull/psd.c#L177)
We should leave it for now for `psd` to work.

File: storage/imx6ull-flash/flashsrv.c (hunk 2 above)
Comment (HBuczynski) [reply storage/imx6ull-flash/flashsrv.c]: Done

File: storage/imx6ull-flash/flashsrv.c [hunk 3]
```diff
@@ -1,637 +1,372 @@
...
-		req = flashsrv_newRequest(fs->port, fs->handler, fs->data);
-
-		if (msgRecv(fs->port, &req->msg, &req->rid) < 0)
-			continue;
-
-		/* TODO: handle umount here? */
-		if (req->msg.type == mtUmount) {
 
-		}
+	memset(msg.i.data, 0, msg.i.size);
```
[the remark this answers is not in the document]
Comment (HBuczynski) [reply storage/imx6ull-flash/flashsrv.c]: Changed malloc to calloc.

**Review by agkaminski — APPROVED** [review]
Answering your question:
- if it's non-critical (i.e. most critical functionality can still be provided) it should be threaded as so. Imho in this case this is indeed non-critical, ofc some erros msg is welcome in such case
- It would be better to provide additional method to do erase, but this can be a feauture of this function if erase function is not available. Imho it's better to pass NULL then allocating 4/64 KB of 0xff data to do the same job.

**Review by Maxez — APPROVED** [review]
LGTM

**Review by agkaminski — APPROVED** [review]

**Pull request merged**